tests/python_tests/test_generate_api.py

```diff
@@ -0,0 +1,309 @@
+    
+    ov_output = pipe.generate(prompt, **config)
+    if config.get('num_return_sequences', 1) > 1:
+        ov_output = ov_output[0]
+
+    if hf_output != ov_output:
+        print(f'hf_output: {hf_output}')
+        print(f'ov_output: {ov_output}')
+
+    assert hf_output == ov_output
+
+
+def stop_criteria_map():
+    # in OpenVINO GenAI this parameter is called stop_criteria,
+    # while in HF it's called early_stopping. 
+    # HF values True, False and "never" correspond to OV GenAI values "EARLY", "HEURISTIC" and "NEVER"
+    return {
+        StopCriteria.NEVER: "never", 
+        StopCriteria.EARLY: True, 
+        StopCriteria.HEURISTIC: False
+    }
+
+
+test_cases = [
+    (dict(max_new_tokens=20), 'table is made of'),  # generation_config, prompt
+    (dict(max_new_tokens=20), '你好！ 你好嗎？'),  # generation_config, prompt
+    (dict(num_beam_groups=3, num_beams=15, num_return_sequences=15, max_new_tokens=20, diversity_penalty=1.0), 'Alan Turing was a'),
+    (dict(num_beam_groups=3, num_beams=15, num_return_sequences=15, max_new_tokens=30, diversity_penalty=1.0), 'Alan Turing was a'),
+    (dict(num_beam_groups=2, num_beams=8, num_return_sequences=8, max_new_tokens=20, diversity_penalty=1.0), 'table is made of'),
+    (dict(num_beam_groups=2, num_beams=8, num_return_sequences=8, max_new_tokens=20, diversity_penalty=1.0), 'The Sun is yellow because'),
+    (dict(num_beam_groups=2, num_beams=8, num_return_sequences=8, max_new_tokens=20, diversity_penalty=1.5), 'The Sun is yellow because'),
+]
+@pytest.mark.parametrize("generation_config,prompt", test_cases)
+@pytest.mark.precommit
+def test_decoding(model_fixture, generation_config, prompt):
+    run_hf_ov_genai_comparison(model_fixture, generation_config, prompt)
+
+
+test_configs = [
+    dict(max_new_tokens=20),
+    dict( max_new_tokens=20, num_beam_groups=3, num_beams=15,diversity_penalty=1.0)
+]
+batched_prompts = [['table is made of', 'They sky is blue because', 'Difference between Jupiter and Mars is that'],
+                   ['hello', 'Here is the longest nowel ever: '],
+                   ['Alan Turing was a', 'return 0', '你好！ 你好嗎？']]
+@pytest.mark.parametrize("generation_config", test_configs)
+@pytest.mark.parametrize("prompts", batched_prompts)
+@pytest.mark.precommit
+def test_multibatch(model_fixture, generation_config, prompts):
+    generation_config['pad_token_id'] = 2
+    run_hf_ov_genai_comparison_batched(model_fixture, generation_config, prompts)
+
+
+prompts = ['The Sun is yellow because', 'Difference between Jupiter and Mars is that', 'table is made of']
+@pytest.mark.parametrize("num_beam_groups", [2, 3, 8])
+@pytest.mark.parametrize("group_size", [5, 3, 10])
+@pytest.mark.parametrize("max_new_tokens", [20, 15])
+@pytest.mark.parametrize("diversity_penalty", [1.0 , 1.5])
+@pytest.mark.parametrize("prompt", prompts)
+@pytest.mark.precommit
+def test_beam_search_decoding(model_fixture, num_beam_groups, group_size, 
```

Comment:
@as-suvorov 
looks like we duplicate beam search testing here and in `test_decoding`. Let's keep only required instances to minimize tests time.